src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -307,6 +325,49 @@ private void nullifyWatchdog() {
             watchdog = null;
         }
     }
+
+    /**
+     * Offers lines in buffer to queue.
+     *
+     * @param cb a buffer to have received text data.
+     * @return the line string. null if no EOL, otherwise buffer is compacted.
+     */
+    private String getLine(CharBuffer cb) {
+        String line = null;
+        int pos = cb.position();
+        cb.flip();
+        for (int i = 0; i < cb.length(); i++) {
+            if (cb.charAt(i) == '\n') {
+                line = getSubSequence(cb, 0, i).toString().trim();
+                cb.position(i + 1);
```

Comment:
Basically, we can ignore the difference of EOL on each OS architecture. Because almost network communication is done by generic rule. e.g. HTTP's basic rule is "EOL is CRLF" defined by RFC2616.
In this case, we can use '\n'(LF) for buffer splitting. After that, '\r'(CR) might be left, but it is removed by String.trim().

I believe this is the same logic as BufferedReader.readLine().
http://hg.openjdk.java.net/jdk7/jdk7/jdk/file/9b8c96f96a0f/src/share/classes/java/io/BufferedReader.java#l313